### PR TITLE
`kci user` commands for user profiles

### PR DESCRIPTION
**Pull request by JenySadadia**
Fixes https://github.com/kernelci/kernelci-core/issues/2016

Added commands to `kci user` for getting user profiles by attributes.
Newly added sample commands:
```
kci user find_users
kci user find_users --limit 2
kci user find_users profile.username=admin
```
Added pagination related optional arguments to `kernelci.cli.base` as they have been used by multiple commands like `kci node find` and `kci user find_groups`.
Optimized `kci node find` command handler.

### CODE REVIEW

File: kernelci/api/__init__.py [hunk 1]
```diff
@@ -125,6 +125,36 @@ def create_node(self, node: dict) -> dict:
...
+    def get_user_profile(self, user_profile_id: str) -> dict:
+        """Get the user profile matching the given profile id"""
+
+    @abc.abstractmethod
+    def get_user_profiles(
+        self, attributes: dict,
+        offset: Optional[int] = None, limit: Optional[int] = None
+    ) -> Sequence[dict]:
+        """Get user profiles that match the provided attributes"""
+
```
Comment (gctucker) [inline kernelci/api/__init__.py:157]: I would suggest to leave this out for now as the API side of user profiles is not ready yet.

File: kernelci/cli/user.py [hunk 2]
```diff
@@ -49,6 +50,54 @@ def _api_call(self, api, configs, args):
...
+       Args.limit, Args.offset
+    ]
+
+    def _api_call(self, api, configs, args):
+        attributes = self._split_attributes(args.attributes)
+        profiles = api.get_user_profiles(attributes, args.offset, args.limit)
+        self._print_json(profiles, args.indent)
+        return True
+
+
```
Comment (gctucker) [inline kernelci/cli/user.py:100]: Likewise, I think we can leave this out for now.  There's already the `whoami` command to get details about the current user.

**Review by gctucker — COMMENTED** [review]
If this PR was reworked to just cover the user groups then I think it could be merged soon.  I believe user profiles aren't required for the "early access" phase and the API side of things is not ready yet.

**Comment by gctucker** [comment]
> If this PR was reworked to just cover the user groups then I think it could be merged soon

Ah I see there's #1972 for user groups.  So let's see if we can get that one merged.

File: kernelci/cli/user.py (hunk 2 above)
Comment (JenySadadia) [reply kernelci/cli/user.py]: Removed `get` command.

**Comment by JenySadadia** [comment]
Tested OK on staging:
```
$ ./kci user find_users --api-config=staging.kernelci.org profile.username=jeny
[
{
"profile": {
"username": "jeny",
"hashed_password": "$2b$12$UjgKqbbJbc3ZM03Hb5aaUO0OJ2KIaqXD.SosWh2U0EEY2tFDQRMi6",
"groups": []
}
}
]
```
```
$ ./kci user find_users --api-config=staging.kernelci.org
[
{
"profile": {
"username": "jeny",
"hashed_password": "$2b$12$UjgKqbbJbc3ZM03Hb5aaUO0OJ2KIaqXD.SosWh2U0EEY2tFDQRMi6",
"groups": []
}
},
{
"profile": {
"username": "testAdmin",
"hashed_password": "$2b$12$7nysdbTVI7dW6AeFDsMxs.bcc9q/k.1FXycqLmmRRV2mArBj74P3W",
"groups": [
{
"id": "649190b01432ccfdb575a250",
"name": "admin"
}
]
}
},
{
"profile": {
"username": "staging.kernelci.org",
"hashed_password": "$2b$12$Dir/xrihCf50kozxKdTX6u5IhnbhldDBjOL446OOChCncwu4K4HIG",
"groups": []
}
},
{
"profile": {
"username": "gtucker",
"hashed_password": "$2b$12$Z2cjFiUlUiJeD8Si93xIV.4EIheWGwtnWNACpiLMJxONnj2npjsSS",
"groups": []
}
},
{
"profile": {
"username": "alex",
"hashed_password": "$2b$12$CCvyfkT1M3mVwCg/FEn.5ezINiQe5PbLcE.FdlXwHIXuyy9maNC5G",
"groups": []
}
},
{
"profile": {
"username": "mg-admin",
"hashed_password": "$2b$12$7sFqfTxfqSqPgz/4yn9tuuUju2G0dHMOpVlbJceik5hil5CwzdHeu",
"groups": []
}
},
{
"profile": {
"username": "rcn",
"hashed_password": "$2b$12$jce8d99FKxULmfBsMSACxuIALgu2NAYG8ufcre5ECARGURv.HYxY6",
"groups": []
}
},
{
"profile": {
"username": "admin",
"hashed_password": "$2b$12$WxssucnPAABBsihbWsU2ceDBQDmpMf8c7gbMuVP861DjTzay5wBf.",
"groups": [
{
"id": "649190b01432ccfdb575a250",
"name": "admin"
}
]
}
}
]

```

**Comment by gctucker** [comment]
@JenySadadia Please don't leak hashed passwords in GitHub comments.  I know it's encrypted and this is a staging database, but this still shouldn't be made available publicly.  It's one of the reasons for having a separate model with the public user profile fields.

**Review by gctucker — APPROVED** [review]

**Comment by gctucker** [comment]
Merging now that all the related API changes have been merged too.

**Pull request merged**